src/main/host/syscall/socket.c

```diff
@@ -1145,6 +1156,38 @@ SysCallReturn syscallhandler_setsockopt(SysCallHandler* sys,
                 sys, socket_desc, optname, optvalPtr, optlen);
             break;
         }
+        case SOL_TCP: {
+            if (descriptor_getType((LegacyDescriptor*)socket_desc) != DT_TCPSOCKET) {
+                errcode = -ENOPROTOOPT;
+                break;
+            }
+
+            /* Shadow doesn't support nagle's algorithm, so shadow always behaves as if TCP_NODELAY
+             * is enabled. Some programs will fail if setsockopt(fd, SOL_TCP, TCP_NODELAY, &1,
+             * sizeof(int)) returns an error, so we treat this as a no-op for compatibility.
+             */
+            if (optname == TCP_NODELAY) {
+                if (optlen < sizeof(int)) {
+                    errcode = -EINVAL;
+                    break;
+                }
+
+                int enable = 0;
+                errcode = process_readPtr(sys->process, &enable, optvalPtr, sizeof(int));
+                if (errcode == 0) {
+                    if (enable) {
+                        // wants to enable TCP_NODELAY
+                        debug("Ignoring TCP_NODELAY");
+                    } else {
+                        // wants to disable TCP_NODELAY
+                        warning("Cannot disable TCP_NODELAY");
```

Comment:
I would add a few more details to the warning message, like "Shadow does not implement Nagle's algorithm so setting `TCP_NODELAY` to 0 will have no effect."

Reply:
Added more context to the warning message.